ocaml/xapi-aux/throttle.mli

```diff
@@ -22,3 +22,33 @@ module Make (_ : SIZE) : sig
 
   val execute : (unit -> 'a) -> 'a
 end
+
+module Batching : sig
+  (** batching delay configuration *)
+  type t
+
+  val make : delay_before:float -> delay_between:float -> t
+  (** [make ~delay_before ~delay_between] creates a configuration,
+    where we delay the API call by [delay_before] once,
+    and then with [delay_between] between each recursive call.
+   *)
+
+  val with_recursive : t -> (('a -> 'b) -> 'a -> 'b) -> 'a -> 'b
+  (** [with_recursive config f arg] calls [f self arg], where [self] can be used
+    for recursive calls.
+
+    A [delay_before] amount of seconds is inserted once, and [delay_between] is inserted between recursive calls:
+    {v
+      delay_before
+      f ...
+        self ...
+         delay_between
+         f ...
+          self ...
+          delay_between
+          f ...
+     v}
+
+    The delays are determined by [config]
+   *)
```

Comment:
I found it difficult to understand what the function does, or how to use it while reading only the interface. I had to read the examples and the implementation to understand that. I found that using the word `loop`
 makes the usage much clearer, including how to control the execution flow, since it's so similar. for example:
```ocaml
let timing = Throttle.Batching.make in
Throttle.Batching.with_loop timing @@ fun loop () ->
...
(loop [@tailcall]) ()
```

If the `[@tailcall]` is needed, I also would mention it in the doccoment

Reply:
I can rename it to `with_recursive_loop`.

Reply:
Also improved docs to mention the exponential increase in delays (capped at `delay_between`) and the tailcall.
The tailcall is not required, but it is useful for the compiler to give a warning if the call is not in the tail position (the compiler always does the tailcall transformation regardless of this annotation).